documentation/coding_guidelines.md

```diff
@@ -227,6 +227,16 @@ impl Foo {
 }
 ```
 
+### Integer Arithmetic
+
+As every integer operation (`+`, `-`, `/`, `*`, etc.) implies edge-cases (e.g. `u64::MAX + 1`),
+we use checked arithmetic instead of directly using the math symbols (`+`, `-`, etc.):
+
+* [checked_](https://doc.rust-lang.org/std/primitive.u32.html#method.checked_add): returns `None` if an overflow has happened, `Some(operation_result)` otherwise
+* [overflowing_](https://doc.rust-lang.org/std/primitive.u32.html#method.overflowing_add): returns the result, potentially wrapping around (e.g. `u64::MAX.overflow_add(1) == 0`), as well as a flag indicating if an overflow has occured or not
+* [wrapping_](https://doc.rust-lang.org/std/primitive.u32.html#method.wrapping_add): similar to overflowing operations, except that it returns the result directly
+* [saturating_](https://doc.rust-lang.org/std/primitive.u32.html#method.saturating_add): if an overflow occurs, the result is kept within the boundary of the type (e.g. `u64::MAX.saturating_add(1) == u64::MAX`)
```

Comment:
this isn't really indicating what to chose or how to chose it, it seems to be a copy of a doc block, perhaps we could pivot more toward recommendations

* choose X if you're not sure what to choose
* if roll over is okay do this
* if roll over would result in safety concerns do this

Reply:
> The sane default is to turn on overflow-checking 

we had this discussion on workplace, and @sblackshear had a pretty good point there. I thought about it more and, I think it is somewhat similar to debug_asserts. For these we deliberately decided not to crash in prod, so why enable them in prod? The same could be thought of integer overflows, but worse: I don't think anyone who wrote `+` or `-` thought about the edge-cases, and decided that they should crash or not crash in prod. Worse, if most integer overflows end up being benign, then we created a lot of DoS attack vectors in prod.

> This generally means "normal" Rust syntax will get you the checked version

I think you mean that it'll get you the "checked + unwrap version" which from my look at the lints, we can avoid in most cases.